crates/forge/tests/e2e/new.rs

```diff
@@ -105,35 +106,73 @@ fn create_new_project_dir_exists_and_empty() {
         .assert()
         .success();
 
-    validate_init(&project_path, false, &Template::BalanceContract);
+    validate_init(&project_path, None, &Template::BalanceContract);
 }
 
 #[test]
 fn init_new_project_from_scarb() {
     let temp = tempdir_with_tool_versions().unwrap();
 
-    SnapboxCommand::new("scarb")
-        .current_dir(&temp)
-        .args(["new", "test_name"])
-        .env("SCARB_INIT_TEST_RUNNER", "starknet-foundry")
-        .env(
-            "PATH",
-            append_to_path_var(snforge_test_bin_path().parent().unwrap()),
-        )
-        .assert()
-        .success();
-
-    validate_init(&temp.join("test_name"), true, &Template::BalanceContract);
+    SnapboxCommand::from_std(
+        ScarbCommand::new()
+            .current_dir(&*temp)
+            .args(["new", "test_name"])
+            .env("SCARB_INIT_TEST_RUNNER", "starknet-foundry")
+            .env(
+                "PATH",
+                append_to_path_var(snforge_test_bin_path().parent().unwrap()),
+            )
+            .command(),
+    )
+    .assert()
+    .success();
+
+    validate_init(
+        &temp.join("test_name"),
+        Some(SnforgeStd::Normal),
+        &Template::BalanceContract,
+    );
 }
 
+#[test]
+#[cfg_attr(not(feature = "scarb_2_9_1"), ignore)]
+fn init_new_project_from_scarb_snforge_std_compatibility() {
+    let temp = tempdir_with_tool_versions().unwrap();
+    let tool_version_path = temp.join(".tool-versions");
+    fs::write(tool_version_path, "scarb 2.11.4").unwrap();
+
+    SnapboxCommand::from_std(
+        ScarbCommand::new()
+            .current_dir(&*temp)
+            .args(["new", "test_name"])
+            .env("SCARB_INIT_TEST_RUNNER", "starknet-foundry")
+            .env(
+                "PATH",
+                append_to_path_var(snforge_test_bin_path().parent().unwrap()),
+            )
+            .command(),
+    )
+    .assert()
+    .success();
+
+    validate_init(
+        &temp.join("test_name"),
+        Some(SnforgeStd::Compatibility),
+        &Template::BalanceContract,
+    );
+}
 pub fn append_to_path_var(path: &Path) -> OsString {
     let script_path = iter::once(path.to_path_buf());
     let os_path = env::var_os("PATH").unwrap();
     let other_paths = env::split_paths(&os_path);
     env::join_paths(script_path.chain(other_paths)).unwrap()
 }
 
-fn validate_init(project_path: &PathBuf, validate_snforge_std: bool, template: &Template) {
+fn validate_init(
+    project_path: &PathBuf,
+    validate_snforge_std: Option<SnforgeStd>,
```

Comment:
nit: Simplified:
```suggestion
    snforge_std: Option<SnforgeStd>,
```

Reply:
I've simplified it in https://github.com/foundry-rs/starknet-foundry/pull/3502 (some error with fixups on my side), please check it out there